include/libtorrent/torrent_info.hpp

```diff
@@ -580,7 +573,7 @@ namespace libtorrent
 		// this is a copy of the info section from the torrent.
 		// it use maintained in this flat format in order to
 		// make it available through the metadata extension
-		boost::shared_array<char> m_info_section;
```

Comment:
this prevents the client from keeping this buffer alive. Given that the client and libtorrent runs in separate threads, it kind of makes sense to actually share ownership. Otherwise the client could have the rug pulled from under its feet at any moment